hnn_core/gui.py

```diff
@@ -0,0 +1,1028 @@
+            )
+
+        if drive_type in [
+                'Evoked',
+                'Poisson',
+                'Rhythmic',
+        ]:
+            drive_boxes.append(drive_box)
+            drive_widgets.append(drive)
+
+        if render:
+            accordion = Accordion(
+                children=drive_boxes,
+                selected_index=len(drive_boxes) -
+                1 if expand_last_drive else None,
+            )
+            for idx, drive in enumerate(drive_widgets):
+                accordion.set_title(idx, drive['name'])
+            display(accordion)
+
+
+def _debug_update_plot_window(variables, _plot_out, plot_type, idx):
+    update_plot_window(variables, _plot_out, plot_type)
+
+
+def update_plot_window(variables, _plot_out, plot_type):
+    _plot_out.clear_output()
+    if not (plot_type['type'] == 'change' and plot_type['name'] == 'value'):
+        return
+
+    with _plot_out:
+        if plot_type['new'] == 'spikes':
+            fig, ax = plt.subplots()
+            variables['net'].cell_response.plot_spikes_raster(ax=ax)
+
+        elif plot_type['new'] == 'current dipole':
+            fig, ax = plt.subplots()
+            # variables['dpls'][0].plot(ax=ax)
+            plot_dipole(variables['dpls'], ax=ax, average=True)
+
+        elif plot_type['new'] == 'input histogram':
+            # BUG: got error here, need a better way to handle exception
+            fig, ax = plt.subplots()
+            variables['net'].cell_response.plot_spikes_hist(ax=ax)
+
+        elif plot_type['new'] == 'PSD':
+            fig, ax = plt.subplots()
+            variables['dpls'][0].plot_psd(fmin=0, fmax=50, ax=ax)
+
+        elif plot_type['new'] == 'spectogram':
+            freqs = np.arange(10., 100., 1.)
+            n_cycles = freqs / 8.
+            fig, ax = plt.subplots()
+            variables['dpls'][0].plot_tfr_morlet(freqs,
+                                                 n_cycles=n_cycles,
+                                                 ax=ax)
+        elif plot_type['new'] == 'network':
+            fig = plt.figure()
+            ax = fig.add_subplot(111, projection='3d')
+            variables['net'].plot_cells(ax=ax)
+
+
+def load_drives(variables, params, log_out, drives_out, drive_widgets,
+                drive_boxes, tstop):
+    """Add drive ipywidgets from params.
+    """
+    variables['net'] = jones_2009_model(params)
+    log_out.clear_output()
+    with log_out:
+        drive_specs = _extract_drive_specs_from_hnn_params(
+            variables['net']._params, list(variables['net'].cell_types.keys()))
+
+        # clear before adding drives
+        drives_out.clear_output()
+        while len(drive_widgets) > 0:
+            drive_widgets.pop()
+            drive_boxes.pop()
+
+        drive_names = sorted(drive_specs.keys())
+        for idx, drive_name in enumerate(drive_names):  # order matters
+            specs = drive_specs[drive_name]
+            print(f"""load drive:
```

Comment:
Triple quotes are for docstrings. Here you should be okay with double quotes.

Reply:
According to the [official document](https://docs.python.org/3.8/tutorial/introduction.html), I guess it should be okay to use triple quotes for multiline strings.

![image](https://user-images.githubusercontent.com/11160442/177399018-37a946a3-6266-4466-bd8d-c56db7c9c191.png)